File: src/com/kraftwerking/datastructuresalgo/leetcode/HouseRobber198.java
package com.kraftwerking.datastructuresalgo.leetcode;

class HouseRobber198 {

  public int rob(int[] nums) {
    int rob1 = 0;
    int rob2 = 0;

    for (int n : nums) {
      int temp = Math.max(n + rob1, rob2);
      rob1 = rob2;
      rob2 = temp;
    }

    return rob2;
  }

  public static void main(String[] args) {
    HouseRobber198 houseRobber198 = new HouseRobber198();
    int[] nums = {1, 2, 3, 1};
    int result = houseRobber198.rob(nums);
    System.out.println(result);
  }
}

File: python/HouseRobber198.py
from typing import List


class Solution(object):
    def rob(self, nums: List[int]) -> int:
        rob1, rob2 = 0, 0

        for n in nums:
            temp = max(n + rob1, rob2)
            rob1 = rob2
            rob2 = temp
        return rob2


def main():
    nums = [1, 2, 3, 1]

    solution = Solution()
    result = solution.rob(nums)
    print(result)


if __name__ == "__main__":
    main()
